### ISSUE TITLE
Client Side Ignore Issue

### ISSUE THREAD

**Issue by Paradigm999** [superzanti/ServerSync#66]
I am successfully syncing with my MC server but my local (client side) VoxelMaps temp files are being deleted regardless of the glob in the FILE_IGNORE_LIST. 
I can't figure out what to put in the FILE_IGNORE_LIST to prevent deletion.

In the serversync-client.cfg file I have these lines:
```
rules {
    S:CONFIG_INCLUDE_LIST <
    >

    S:FILE_IGNORE_LIST <
        *.zip
        **/*.zip
    >
}
```

Here is the output in the serversync-detailed.log:
```
<------> Getting files <------>
Getting mods...
Recieved server file tree
Accepting client mods! Added client mods to server list
Recieved client only files
Ignoring [*.zip, **/*.zip]
<------> Starting Update Process <------>
..... Some lines removed for brevity
<------> Starting deletion process <------>
..... Some lines removed for brevity
Checking client's -1,-1.zip
-1,-1.zip Does not match...Attempting to delete
<>-1,-1.zip deleted
```

The files in question are located in the folder:
`mods\VoxelMods\voxelMap\cache\serverip~colon~25565\Overworld (dimension 0)`
I tried to put the whole folder name in the cfg file but that didn't work either. I'm not sure if it should have quotes or the spaces and colons should be escaped or if it can go in as-is.

I am running Windows 10 and java version:

> java version "1.8.0_121"
> Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
> Java HotSpot(TM) Client VM (build 25.121-b13, mixed mode, sharing)

Please let me know if you need any more information.

**Comment by rheimus** [comment]
I get a cheeky suspicion that you might be adding these globs to the serversync-client.cfg that is in the servers directory:
````
server/
    config/
        serversync/
            serversync-client.cfg
````

This is actually a bug that I have been terribly lazy at removing :). If this is the case then the appropriate globs just need to be copied to the serversync-client.cfg that is contained in the client directory.

I should really provide a template for reporting so it's easier to know what to provide, my bad, only things missing are:
- Minecraft version
- Serversync version

Thank you for the detailed report though.

What your two globs are actually doing:
````
*.zip // anything with the .zip extension contained in the base minecraft directory
      // MinecraftDir/*.zip
````
````
**/*.zip // Anything in any sub directory (recursive) of minecraft with a zip extension
         // Minecraft/(any sub | any depth)/*.zip
         // Note be careful with this one as it is probably not what you want 
         // (e.g. will ignore all flan content packs in /flan)
         // Essentially this one is saying any zip file as long as its in a sub directory somewhere
````

Theres a nice glob testing tool here: [Globtester](http://www.globtester.com/)

**Comment by rheimus** [comment]
Actually come to think of it, I may not have implemented the globbing in the deletion phase.

I'll check this out tonight.

**Comment by Paradigm999** [comment]
I tried editing the cfg file that is located in %appdata%\.minecraft\config\serversync\serversync-client.cfg
I also tried editing the client config on the MC server. Editing the file on the server actually merged the differences in the two files and saved the changes to the client.

I added the glob **/*.zip as a nuclear catch-all that should have worked. When it didn't, I filed the issue here.

The problem with adding these files to the ignore list is that they are auto generated and I cannot predict the names. My other computer has almost a hundred of them. They are a cache for a mini map. If they are deleted, I lose my map. 


My Minecraft version is 1.11.2 with Forge 1.11.2-13.20.0.2228
Serversync version is 2.6.11

**Comment by rheimus** [comment]
The client is indeed still using the old matching system.

Shall be fixed in the next update

**Comment by rheimus** [comment]
Fixed in 2.6.12

**Comment by Paradigm999** [comment]
It works great! Thank you for the quick turnaround!

I was able to go back to a safer ignore glob of mods/VoxelMods/**/*.zip
